src/summary.cpp

```diff
@@ -49,54 +50,71 @@ void Summary::add_thread()
     thread_count_++;
 }
 
-std::string pretty_print_bytes(int trace_size)
+std::string pretty_print_bytes(std::size_t trace_size)
 {
     double result_size = trace_size;
     int unit = 0;
-    std::string units[5] = { "B", "KiB", "MiB", "GiB", "TiB" };
 
-    for (; result_size > 1024; unit++, result_size /= 1024)
-        ;
+    std::ostringstream out;
```

Comment:
the variable was declared at the right spot before this change. Always declare variables as late as possible and preferable only, when you can initialize them.